region/wreckedship/main.json

```diff
@@ -3632,56 +3798,75 @@
-                    {"or": [
-                      {"and": [
-                        "canMidairShinespark",
-                        {"canComeInCharged": {
-                          "fromNode": 1,
-                          "framesRemaining": 30
-                        }},
-                        {"shinespark": {
-                          "frames": 39,
-                          "excessFrames": 10
-                        }}
-                      ]},
-                      {"and": [
-                        {"or": [
-                          "canShinechargeMovement",
-                          "canWaterShineCharge"
-                        ]},
-                        {"canComeInCharged": {
-                          "fromNode": 1,
-                          "framesRemaining": 180
-                        }},
-                        {"shinespark": {
-                          "frames": 7,
-                          "excessFrames": 2
-                        }}
-                      ]}
-                    ]}
+                    "canWaterShineCharge",
+                    {"shinespark": {
+                      "frames": 7,
+                      "excessFrames": 2
+                    }}
+                  ],
+                  "devNote": "The 17 submerged tiles behave as 7 tiles with high run speed. The 3 steep down tiles behave as approximately 1."
+                },
+                {
+                  "name": "Shinespark, Come in Charged",
+                  "notable": false,
+                  "entranceCondition": {
+                    "comeInShinecharged": {
+                      "framesRequired": 30
+                    }
+                  },
+                  "requires": [
+                    "canMidairShinespark",
+                    {"shinespark": {
+                      "frames": 39,
+                      "excessFrames": 10
+                    }}
                   ]
                 },
                 {
                   "name": "Exterior Speedjump",
                   "notable": false,
+                  "entranceCondition": {
+                    "comeInRunning": {
+                      "speedBooster": true,
+                      "minTiles": 3
+                    }
+                  },
                   "requires": [
                     "canSuitlessMaridia",
-                    {"adjacentRunway": {
-                      "fromNode": 1,
-                      "usedTiles": 3,
-                      "inRoomPath": [ 1, 2 ],
-                      "physics": ["normal"]
-                    }},
-                    "SpeedBooster",
+                    "canCarefulJump",
                     {"or": [
                       "canWalljump",
                       "HiJump"
                     ]}
                   ],
-                  "note": "A bit finicky for shorter runways, but exactly 3 tiles works."
+                  "note": "Requires at least 3 tiles to work, but some longer lengths don't work."
```

Comment:
Exactly 3 tiles looked like it doesn't work, I needed a few more pixels.  I didn't hit any of the longer length anti-trickyDashJumps; not sure that that needs to be mentioned.

Reply:
The old way to define runways was very ambiguous, and I've noticed a lot of variation based on who/when it was added. With the new schema we adopted Bobbob's way of calculating it, which is: tiles of movement that Samus has. So 3 tiles of runway in the next room is the same as 3 tiles + an open end (including the door cap), so Samus can traverse 48 pixels before entering the room.